File: algorithms/Sorting/Medium/BucketSort/VSDBucketSort.py

def VSDBucketSort(arr,n):
    mini=min(arr)
    maxi=max(arr)
    bucket=[[] for i in range(n)]
    for i in range(n):
        j=n*(arr[i]-mini)//(maxi+1-mini)
        bucket[j].append(arr[i])
    i=0
    for j in range(n):
        temp=bucket[j]
        temp.sort()
        for k in range(len(temp)):
            arr[i]=temp[k]
    
    print(*arr,sep=" ")

if __name__=="__main__":
    n=7
    arr=[2,24,12,96, 456, 899,34]
    VSDBucketSort(arr,n)

File: algorithms/Sorting/Medium/BucketSort/VSDBucketSort.cpp

#include<iostream>
#include<vector>
#include<bits/stdc++.h>

using namespace std;
int min(int arr[],int n){
    int mini=arr[0];
    for(int i=0;i<n;i++){
        if(arr[i]<mini)
            mini=arr[i];
    }
    return mini;
}
int max(int arr[],int n){
    int maxi=arr[0];
    for(int i=0;i<n;i++){
        if(arr[i]>maxi)
            maxi=arr[i];
    }
    return maxi;
}

void bucketSort(int arr[],int n){
    int mini=min(arr,n);
    int maxi=max(arr,n);
    vector<vector<int>> bucket(n);

    for(int i=0;i<n;i++){
        int j=n*(arr[i]-mini)/(maxi+2-mini);
        bucket[j].push_back(arr[i]);
    }
    int i=0;
    for(int j=0;j<n;j++){
        vector<int> bj=bucket[j];
        sort(bj.begin(),bj.end());
        for(int k=0;k<bj.size();k++){
            arr[i++]=bj.at(k);
        }
    }
}

int main(){

    int arr[]={5,2, 9, 12, 54, 31, 65, 31, 45, 65};
    int n=sizeof(arr)/sizeof(arr[0]);
    bucketSort(arr,n);
    for(int i=0;i<n;i++){
        cout<<arr[i]<<" ";
    }
    cout<<endl;
    return 0;
}